testing/citest/spinnaker_testing/gcs_pubsub_trigger_agent.py

```diff
@@ -71,131 +71,128 @@ def new_gcs_pubsub_trigger_operation(
 class BaseGcsPubsubTriggerOperation(base_agent.AgentOperation):
   """Specialization for base gcs pubsub trigger operations.
   """
-  def __init__(self, title, gcs_pubsub_agent, max_wait_secs=None):
-    self.__title = title
-    self.__agent = gcs_pubsub_agent
-    super(BaseGcsPubsubTriggerOperation, self).__init__(
-        title, gcs_pubsub_agent, max_wait_secs=max_wait_secs)
+  def __init__(self, title, gcs_pubsub_agent, gate_agent, **kwargs):
+    """Construct operation
+
+    Args:
+      gcs_pubsub_agent: [GcsFileUploadAgent] agent to perform operation
+      gate_agent: [GateAgent] agent used for monitoring status
+    """
     if (not gcs_pubsub_agent
         or not isinstance(gcs_pubsub_agent, GcsFileUploadAgent)):
-      raise TypeError('agent is not a GcsFileUploadAgent: '
+      raise TypeError('gcs_pubsub_agent is not a GcsFileUploadAgent: '
                       + gcs_pubsub_agent.__class__.__name__)
+    if (not gate_agent or not isinstance(gate_agent, GateAgent)):
+      raise TypeError('gate_agent is not a GateAgent: '
+                      + gate_agent.__class__.__name__)
+    super(BaseGcsPubsubTriggerOperation, self).__init__(
+        title, gate_agent, **kwargs)
+    self.__pubsub_agent = gcs_pubsub_agent
 
   def export_to_json_snapshot(self, snapshot, entity):
     snapshot.edge_builder.make_mechanism(
-        entity, 'Gcs Pubsub Agent', self.agent)
+        entity, 'Gcs Pubsub Agent', self.__pubsub_agent)
     super(BaseGcsPubsubTriggerOperation, self).export_to_json_snapshot(
         snapshot, entity)
 
   def execute(self, agent=None):
-    status = self._do_execute(self.agent)
+    status = self._do_execute(self.__pubsub_agent)
```

Comment:
The log message following this line uses `self.agent`, but we changed the reference here.